itu-minitwit-golang/internal/service/handle_login.go

```diff
@@ -0,0 +1,32 @@
+package service
+
+import (
+	"itu-minitwit/internal/utils"
+	"net/http"
+
+	"github.com/gin-contrib/sessions"
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+func HandleLogin(c *gin.Context, username string, password string) string {
```

Comment:
The return-type should be a "error" if you want to return it
i would change return signature to (err error, msg string)

And then for case "invalid username" and "invalid password" return (nil, "invalid username/password")
And for case db error return something like (err, "Internal error")
as the idea with the message is for the user to see, and they are not interested in seeing "database error"